avocado/core/parser.py

```diff
@@ -68,6 +68,7 @@ def __init__(self):
         self.application.add_argument('--show', action="store",
                                       type=lambda value: value.split(","),
                                       metavar="STREAM[:LVL]",
+                                      nargs='?',
```

Comment:
Well this solves the issue, indeed. The only problem I have is it makes the argument optional in the help string `--show [STREAM[:LVL]]`. But it handles the default properly (`avocado --show --config avocado.cfg run passtest`) I like it.

Now @apahim would you please add the same fix for all top-level options (the only missing is `--config`, if I'm not wrong.)